Android/Tapp/app/src/main/java/app/anish/com/tapp/dialogs/LinkedInWebViewLoginDialog.java

```diff
@@ -0,0 +1,294 @@
+                    String msg = "State token doesn't match. Possible CSRF Attach";
+                    dismissDialogInError(msg);
+                    return true;
+                }
+
+                // If user doesn't allow authorization to our application,
+                // the authorizationToken will be null
+                String authorizationToken = uri.getQueryParameter(RESPONSE_TYPE_VALUE);
+                if (authorizationToken == null) {
+                    dismissDialogInError("The user didn't allow authorization");
+                    return true;
+                }
+
+                Log.i(LOG_TAG, "Auth token received " + authorizationToken);
+                String accessTokenUrl = getAccessTokenUrl(authorizationToken);
+
+                // Make the request in an Async Task
+                new PostRequestAsyncTask().execute(accessTokenUrl);
+            } else {
+                // default behaviour
+                String url = uri.toString();
+                Log.i(LOG_TAG, "Webview redirecting to : " + uri);
+                mWebView.loadUrl(url);
+            }
+
+            return true;
+        }
+    }
+
+    private class PostRequestAsyncTask extends AsyncTask<String, Void, Throwable> {
+
+        private volatile AccessToken accessToken;
+
+        @Override
+        protected void onPreExecute() {
+            if (mProgressDialog == null || !mProgressDialog.isShowing()) {
+                mProgressDialog = ProgressDialog.show(getContext(), "", getContext().getString(R.string.loading), true);
+            }
+        }
+
+        @Override
+        protected void onPostExecute(Throwable throwable) {
+            if (mProgressDialog != null && mProgressDialog.isShowing()) {
+                mProgressDialog.dismiss();
+            }
+
+            if (throwable != null) {
+                dismissDialogInError(throwable.getMessage());
+            } else {
+                dismissDialogInSuccess(accessToken);
+            }
+        }
+
+        @Override
+        protected Throwable doInBackground(String... urls) {
+            if (urls.length > 0) {
+                String url = urls[0];
+                try {
+                    accessToken = getLinkedinToken(url);
+                    Log.d("LINKEDIN TOKEN", "LinkedIn Token is " + accessToken.toString());
+                    return null;
+                } catch (IOException e) {
+                    return new Exception("Error Http response " + e.getLocalizedMessage());
+                } catch (JSONException e) {
+                    return new Exception("Error Parsing Http response " + e.getLocalizedMessage());
+                }
+            }
+
+            return new Exception("No urls provided to task");
+        }
+
+        /**
+         * Extracts token from URL and saves it to Shared Preferences
+         * @param url
+         */
+        private AccessToken getLinkedinToken (String url) throws IOException, JSONException{
```

Comment:
It depends upon your purpose. If you're trying to get something out the door because you have a deadline to hit and you can't spend time learning to use tools that will help you long-term, then whatever approach is fastest for you will be the right one to use. However, if you're wanting to use this project as a vehicle to learn how to make great Android Apps and be a productive Android developer, then I would suggest using the tools that will make you productive.

You don't have to use the ones I mentioned, but in my experience, when you use them, the amount of code you have to write is reduced and the approach can be scaled to situations in which you have many pojo classes. The best opportunity to learn to use these tools is in uncomplicated situations. Therefore, I think this is a really good place to get started with them.